test/src/test/java/jenkins/model/JenkinsTest.java

```diff
@@ -771,6 +774,14 @@ public void testTrimLabelsRetainsLabelExpressions() throws Exception {
         assertThat(j.jenkins.getLabels().contains(l), is(true));
     }
 
+    @Test
+    public void reloadShouldNotSaveConfig() throws Exception {
+        Path configXml = j.jenkins.getRootDir().toPath().resolve("config.xml");
+        FileTime lastModifiedTime = Files.getLastModifiedTime(configXml);
+        j.jenkins.reload();
+        assertThat("config.xml should not have been saved", Files.getLastModifiedTime(configXml), is(lastModifiedTime));
```

Comment:
https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/nio/file/attribute/FileTime.html#equals(java.lang.Object) is this reliable across platforms?

Might be more reliable to use a `SaveableListener` as a `@TestExtension`.